### PR TITLE
ENH Add python code for notebook used in the parallel coordinate plot video

### LINKED ISSUE #447: Add video showing how to select a range of results in a parallel coordinate plot
Following Guillaume's suggestion in [this forum comment](https://mooc-forums.inria.fr/moocsl/t/video-to-explain-how-to-use-the-interactive-parallel-coordinates-plot/2548/4), a video after [this notebook](https://inria.github.io/scikit-learn-mooc/python_scripts/parameter_tuning_randomized_search.html) showing how to select a range of results in a parallel coordinate plot may suffice to complement the explanation on how to use this tool.

This may also improve the scores for Quiz M3.2 Q4 and Q5, which are both below 70%.

### PR DESCRIPTION
**Pull request by ArturoAmorQ**
Fixes #447.

This PR partially moves the contents of the [Hyperparameter tuning by randomized-search notebook](https://inria.github.io/scikit-learn-mooc/python_scripts/parameter_tuning_randomized_search.html) to a new notebook that was used in the corresponding parallel coordinate plot video. The content that was suppressed is replaced with a brief discussion on the stability of the parameters found for the top performing models. This may hopefully serve as motivation for the need of a way to visualize the scores.

The idea is that the video and notebook will be displayed simultaneously within a split webpage. This may require FUN changes.

### CODE REVIEW

File: python_scripts/video_parallel_plot.py [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+        return param_name.rsplit("__", 1)[1]
+    return param_name
+
+
+# %%
+cv_results = cv_results.rename(shorten_param, axis=1)
+cv_results.head()
+
+# %% [markdown]
+# As we have more than 2 parameters in our randomized-search, we
```
Comment (glemaitre) [inline python_scripts/video_parallel_plot.py:37]: It is a bit weird here because we say that we should not do a pairwise comparison but we show such a plot and do not explain what is it. I assume that we should at least make an analysis of the plot?

File: python_scripts/parameter_tuning_randomized_search.py [hunk 2]
```diff
@@ -238,66 +255,17 @@ def shorten_param(param_name):
...
-# selection by clicking once again on the same axis.
-#
-# We also observe that it is not possible to select the highest performing
-# models by selecting lines of on the `max_bins` axis with tick values between
-# 1 and 3.
-#
-# The other hyper-parameters are not very sensitive. We can check that if we
-# select the `learning_rate` axis tick values between -1.5 and -0.5 and
-# `max_bins` tick values between 5 and 8, we always select top performing
-# models, whatever the values of the other hyper-parameters.
```
Comment (ogrisel) [inline python_scripts/parameter_tuning_randomized_search.py:296]: Are you sure you want to remove this from `python_scripts/parameter_tuning_randomized_search.py `?

If you add it only to `python_scripts/video_parallel_plot.py `, this notebook will no longer be reachable from the navigation menus.

I think it's import ant to keep it as part of the interactive notebooks from the module. I would insert the video in the sequence right after `python_scripts/parameter_tuning_randomized_search.py` so that readers:

- first see the interactive plots and play with it intuitively but probably miss how to use it correctly,
- then watch the video;
- the go back to the interactive plot to check they can select a range by clicking on the axis.

Comment (ogrisel) [reply python_scripts/parameter_tuning_randomized_search.py:296]: Or alternatively we can keep the split in two notebooks and insert the page with the new video between them but please adjust the `_toc.yml` file as part of this PR so that we have the big picture before merging the PR.

File: python_scripts/parameter_tuning_randomized_search.py [hunk 3]
```diff
@@ -238,66 +255,17 @@ def shorten_param(param_name):
...
+# overlap between each other, meaning that indeed, the set of parameters
+# leading to the best generalization performance is not unique.
 
 # %% [markdown]
 #
-# In this notebook, we have seen how randomized search offer a valuable
+# In this notebook, we saw how a randomized search offers a valuable
 # alternative to grid-search when the number of hyperparameters to tune is more
 # than two. It also alleviates the regularity imposed by the grid that might be
 # problematic sometimes.
```
Comment (ogrisel) [inline python_scripts/parameter_tuning_randomized_search.py:271]:
```suggestion
# problematic sometimes.
#
# In the following, we will see how to use interactive plotting tools to explore
# the results of large hyper-parameter search sessions and gain some
# insights on range of parameter values that lead to the highest performing
# models and how different hyper-parameter are coupled or not.
```

File: python_scripts/video_parallel_plot.py [hunk 4]
```diff
@@ -0,0 +1,129 @@
...
+# %%
+def shorten_param(param_name):
+    if "__" in param_name:
+        return param_name.rsplit("__", 1)[1]
+    return param_name
+
+
+# %%
+cv_results = cv_results.rename(shorten_param, axis=1)
+cv_results.head()
```
Comment (ogrisel) [inline python_scripts/video_parallel_plot.py:34]: I find it more informative to have the default display of dataframes instead of `.head()`.

```suggestion
cv_results
```

File: python_scripts/video_parallel_plot.py [hunk 5]
```diff
@@ -0,0 +1,129 @@
...
+#
+# The other hyperparameters are not very sensitive. We can check that if we
+# select the `learning_rate` axis tick values between -1.5 and -0.5 and
+# `max_bins` tick values between 5 and 8, we always select top performing
+# models, whatever the values of the other hyper-parameters.
+
+# %% [markdown]
+#
+# In this notebook, we saw how to visualize a randomized search when the number
+# of hyperparameters to tune is more than two.
```
Comment (ogrisel) [inline python_scripts/video_parallel_plot.py:129]:
```suggestion
# In this notebook, we saw how to interactively explore the results of 
# large a randomized search with multiple interacting hyper-parameters
# In particular we observed that some hyper-parameters have very little
# impact on the cross-validation score, while others have to be adjusted
# within a specific range to get models with good predictive accuracy.
```

File: python_scripts/video_parallel_plot.py (hunk 1 above)
Comment (ArturoAmorQ) [reply python_scripts/video_parallel_plot.py]: Thanks, done!

File: python_scripts/parameter_tuning_randomized_search.py (hunk 2 above)
Comment (ArturoAmorQ) [reply python_scripts/parameter_tuning_randomized_search.py:296]: Thanks, done!

File: jupyter-book/tuning/parameter_tuning_parallel_plot_video.md [hunk 6]
```diff
@@ -0,0 +1 @@
+# 🎥 Hyperparameter tuning visualization
```
Comment (ogrisel) [inline jupyter-book/tuning/parameter_tuning_parallel_plot_video.md:1]:
```suggestion
# 🎥 Hyperparameter tuning visualization

TODO: insert video player here once ready
```

File: python_scripts/parameter_tuning_parallel_plot.py [hunk 7]
```diff
@@ -0,0 +1,134 @@
+# %% [markdown]
+# # Hyperparameter tuning visualization
```
Comment (ogrisel) [inline python_scripts/parameter_tuning_parallel_plot.py:2]:
```suggestion
# # Interactive exploration of hyper-parameter search results
```

**Review by ogrisel — APPROVED** [review]
LGTM with the title renaming suggestion above.

File: python_scripts/parameter_tuning_parallel_plot.py (hunk 7 above)
Comment (ArturoAmorQ) [reply python_scripts/parameter_tuning_parallel_plot.py:2]: I am just worried about changing the title here but not inside the video, though I like your suggestion the best.  Do you think it's ok to have this small difference?

Comment (ogrisel) [reply python_scripts/parameter_tuning_parallel_plot.py:2]: I think it's fine if the in-video title is not exactly the same.
